src/controllers/phishing/phishing.ts

```diff
@@ -0,0 +1,161 @@
+      this.#blacklist = new Set([
+        ...storedPhishingDetection.metamaskBlacklist,
+        ...storedPhishingDetection.phantomBlacklist
+      ])
+    }
+    await this.#update(storedPhishingDetection)
+    this.isReady = true
+
+    this.emitUpdate()
+  }
+
+  async #update(
+    storedPhishingDetection: {
+      timestamp: number
+      metamaskBlacklist: string[]
+      phantomBlacklist: string[]
+    } | null
+  ) {
+    this.updateStatus = 'LOADING'
+
+    const headers = {
+      Accept: 'application/vnd.github.v3.+json'
+    } as any
+    const results = await Promise.allSettled([
+      this.#fetch(METAMASK_BLACKLIST_URL, headers)
+        .then((res) => res.json())
+        .then((metadata) => fetch(metadata.download_url))
+        .then((rawRes) => rawRes.json())
+        .then((data) => data.blacklist)
+        .catch(() => []),
+      this.#fetch(PHANTOM_BLACKLIST_URL, headers)
+        .then((res) => res.json())
+        .then((metadata) => fetch(metadata.download_url))
+        .then((res) => res.text())
+        .then((text) => jsYaml.load(text))
+        .then((data: any) => (data && data.length ? data.map((i: { url: string }) => i.url) : []))
+        .catch(() => [])
+    ])
+
+    let [metamaskBlacklist, phantomBlacklist] = results.map((result) =>
+      result.status === 'fulfilled' ? result.value || [] : []
+    )
+
+    if (metamaskBlacklist && phantomBlacklist) {
+      const timestamp = Date.now()
+      await this.#storage.set('phishingDetection', {
+        timestamp,
+        metamaskBlacklist: metamaskBlacklist || [],
+        phantomBlacklist: phantomBlacklist || []
+      })
+      this.#latestStorageUpdate = timestamp
+    } else if (storedPhishingDetection && !this.#latestStorageUpdate) {
+      this.#latestStorageUpdate = storedPhishingDetection.timestamp
+    }
+
+    if (storedPhishingDetection) {
+      metamaskBlacklist = metamaskBlacklist || storedPhishingDetection.metamaskBlacklist
+      phantomBlacklist = phantomBlacklist || storedPhishingDetection.phantomBlacklist
+    }
+
+    this.#blacklist = new Set([...metamaskBlacklist, ...phantomBlacklist])
+    this.updateStatus = 'INITIAL'
+  }
+
+  async updateIfNeeded() {
+    if (this.updateStatus === 'LOADING') return
+    const sixHoursInMs = 6 * 60 * 60 * 1000
+
+    if (this.#latestStorageUpdate && Date.now() - this.#latestStorageUpdate < sixHoursInMs) return
+    const storedPhishingDetection = await this.#storage.get('phishingDetection', null)
+
+    if (Date.now() - storedPhishingDetection.timestamp >= sixHoursInMs) {
+      await this.#update(storedPhishingDetection)
+    }
+  }
+
+  async getIsBlacklisted(url: string) {
+    this.emitUpdate()
+    await this.initialLoadPromise
+
+    try {
+      const hostname = new URL(url).hostname
+
+      // blacklisted if it has `ambire` in the hostname but it is not a pre-approved ambire domain
+      if (hostname.includes('ambire') && !hostname.includes('ambire.com')) return true
```

Comment:
I'm pretty sure that this could be bypassed
![image](https://github.com/user-attachments/assets/f33304a4-178b-4eb2-b383-26b41dbecafb)